testing/src/scenario/state.py

```diff
@@ -747,9 +747,9 @@ def _run(self) -> int:
 class Mount(_max_posargs(0)):
     """Maps local files to a :class:`Container` filesystem."""
 
-    location: str | PurePosixPath
+    location: str | pathlib.PurePosixPath
     """The location inside of the container."""
```

Comment:
Do you know why we need `PurePosixPath` here btw? Anything worth documenting (possibly in a future PR). Not a blocker, just stood out to me, and I'm curious

Reply:
This pre-dates me, but I assume we're using a pure path here because the path doesn't actually exist on the testing system, only instead of the (mock) container. POSIX because we're assuming a non-Windows image.

Or if the question is why not just `str` I assume it's to give people more flexibility, for those that like a plain `/charm-stuff/blah.txt` vs. those that like only using path objects. We could take a stand on that and only accept one, I imagine, but not until the next major version at this point.

Reply:
Sorry for the ambiguity -- your first paragraph answers my question. Thanks. No documentation update should be needed as users will be aware that it's a mock container, and the `PurePosixPath` will rebel if people accidentally try to use concrete path operations. Resolving this.